k9mail/src/main/java/com/fsck/k9/view/MessageHeader.java

```diff
@@ -342,6 +346,23 @@ public void populate(final Message message, final Account account) {
         }
     }
 
+    public boolean showSender(Message message) {
+        Address[] from = message.getFrom();
+        Address[] sender = message.getSender();
+        if(from == null || sender == null) {
+            return false;
+        }
+        if(from.length != sender.length) {
+            return false;
```

Comment:
Why return false here? Are you not just trying to check if the arrays are different?